core/src/main/java/io/questdb/griffin/FunctionParser.java

```diff
@@ -691,49 +777,7 @@ private Function createFunction(
                         }
 
                         sigArgTypeScore += overloadDistance;
-                        // Overload with cast to higher precision
                         overloadPossible = overloadDistance != ColumnType.OVERLOAD_NONE;
-
```

Comment:
these are cast hacks we had, they are ad-hoc. I moved them all into a single table, controlled by `ColumnType.OVERLOAD_PRIORITY` - this is the name of the PR originally